File: 215-KthLargestElementInAnArray/215-KthLargestElementInAnArray.java
// Last updated: 8/13/2025, 8:21:16 PM
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> p = new PriorityQueue();
        for(int i=0;i<k;i++){
            p.add(nums[i]);
        }
        for(int i=k;i<nums.length;i++){
            if(p.peek()<nums[i]){
                p.remove();
                p.add(nums[i]);
            }
        }
        return p.peek();
    }
}

File: 215-KthLargestElementInAnArray/215-KthLargestElementInAnArray.py
# Last updated: 8/13/2025, 8:21:11 PM
import heapq
class Solution:
    def findKthLargest(self, arr: List[int], k: int) -> int:
        heap = []
        
        for i in range(len(arr)) :
            if len(heap) < k : 
                heapq.heappush(heap , arr[i])
            else : 
                if arr[i] > heap[0] :
                    heapq.heappop(heap)
                    heapq.heappush(heap , arr[i])
        return heapq.heappop(heap)